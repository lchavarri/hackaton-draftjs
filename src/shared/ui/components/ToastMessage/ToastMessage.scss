@import '../../assets/variables';

.ui.toast.message {
  width: rem(344px);
  background-color: $black;
  color: white;
  min-height: rem(40px);
  border-radius: rem(4px);
  box-shadow: rem(0 2px 4px 0) rgba(0, 0, 0, 0.25);
  margin-bottom: rem(1px);
  left: rem(33%);
  z-index: 2;
  position: absolute;
  top: auto;
  margin-top: rem(-15px);
  padding: rem(12px 19px);

  .min & {
    margin-top: rem(-11px);
  }

  .icon.close {
    cursor: pointer;
    display: flex;
    justify-content: center;
    align-items: center;
    top: 0;
    right: rem(-12px);
    width: rem(32px);
    height: rem(15px);
    color: white;
    position: relative;
    float: right;
    opacity: 0.9;
    z-index: 1;

    &:before {
      content: '';
    }

    svg {
      width: rem(10px);
      height: rem(16px);
    }
  }

  p {
    font-size: rem(11px);
  }
}
