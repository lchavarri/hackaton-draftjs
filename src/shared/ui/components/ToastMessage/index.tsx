import React, { useState, useEffect } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/pro-light-svg-icons';

import './ToastMessage.scss';

export enum ToastType {
  TIMED_TOAST = 'TIMED_TOAST',
  DISMISSABLE_TOAST = 'DISMISSABLE_TOAST'
}

enum ToastTypeTime {
  TIMED_TOAST = 2750,
  DISMISSABLE_TOAST = 5000
}

type Props = {
  message: string;
  type: ToastType;
  onClose?: any;
  setVisible?: any;
  visible: boolean;
};

export default function ToastMessage(props: Props) {
  const [displayTimeout, setDisplayTimeout] = useState<any>();
  const [visible, setVisible] = useState(false);
  const { message, type, onClose } = props;

  useEffect(() => {
    if (props.visible) {
      setVisible(true);
      if (props.setVisible) {
        props.setVisible();
      }
      if (displayTimeout) {
        clearTimeout(displayTimeout);
        setDisplayTimeout(null);
      }
      setDisplayTimeout(
        setTimeout(() => {
          setVisible(false);
        }, ToastTypeTime[type])
      );
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [props.visible]);

  const handleClose = () => {
    setVisible(false);
    if (onClose) {
      onClose();
      if (props.setVisible) {
        props.setVisible();
      }
    }
  };

  const renderClose = () => {
    const { type } = props;
    if (type === ToastType.TIMED_TOAST) return null;

    return (
      <i className="icon close" onClick={handleClose} aria-hidden="true">
        <FontAwesomeIcon icon={faTimes} />
      </i>
    );
  };

  if (!visible) {
    return null;
  }

  return (
    <div className="ui toast message">
      {renderClose()}
      <div className="content">
        <p>{message}</p>
      </div>
    </div>
  );
}
