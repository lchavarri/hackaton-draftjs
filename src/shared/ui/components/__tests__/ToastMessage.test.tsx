import React from 'react';
import { shallow, mount, ReactWrapper } from 'enzyme';
import ToastMessage from '../ToastMessage';

describe('ToastMessage component', () => {
  let component: ReactWrapper;
  let props: any;

  const defaultProps = {};
  const getComponent = () => <ToastMessage {...props} />;
  const wrap = () => shallow(getComponent());
  const wrapDeep = () => mount(getComponent());

  beforeEach(() => {
    props = defaultProps;
    component = wrap();
  });

  it('should match component snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
