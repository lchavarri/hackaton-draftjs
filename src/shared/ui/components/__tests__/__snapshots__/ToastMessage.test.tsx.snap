// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastMessage component should match component snapshot 1`] = `""`;
